cluster/src/main/java/io/atomix/cluster/impl/DefaultClusterMembershipService.java

```diff
@@ -297,20 +313,18 @@ private synchronized void deactivateMember(Member member) {
   @Override
   public CompletableFuture<ClusterMembershipService> start() {
     if (started.compareAndSet(false, true)) {
-      broadcastService.addListener(broadcastListener);
+      locationProvider.addListener(locationEventListener);
       LOGGER.info("{} - Member activated: {}", localMember.id(), localMember);
       localMember.setState(State.ACTIVE);
       members.put(localMember.id(), localMember);
       messagingService.registerHandler(HEARTBEAT_MESSAGE, this::handleHeartbeat, heartbeatExecutor);
 
       ComposableFuture<Void> future = new ComposableFuture<>();
-      broadcastIdentity();
       sendHeartbeats().whenComplete((r, e) -> {
         future.complete(null);
       });
 
       heartbeatFuture = heartbeatScheduler.scheduleWithFixedDelay(() -> {
-        broadcastIdentity();
```

Comment:
isn't this going to break discovery of data nodes unless multicast is enabled?

Reply:
ping @kuujo

Reply:
No. When multicast was disabled `broadcastIdentity()` just did nothing. It's really the higher level `ClusterMembershipService` TCP heartbeats that notify other nodes about membership information, and without that node discovery would not work.